src/main/resources/org/broadinstitute/hellbender/utils/plotter/TangentResultsPlotting.R

```diff
@@ -0,0 +1,168 @@
+print(opt)
+save(opt, file="debug.RData")
+
+sample_name=opt[["sample_name"]]
+tn_file=opt[["targets_file"]]
+pre_tn_file=opt[["pre_tn_file"]]
+seg_file=opt[["seg_file"]]
+output_file=opt[["output_dir"]]
+log_input=as.logical(opt[["log2_input"]])
+
+# Use a function for debugging purposes
+create_tangent_plots_file = function(sample_name, tn_file, pre_tn_file, seg_file, output_dir, log_input) {
+	# Read in file and extract needed data
+	seg = read.table(seg_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+	tn = read.table(tn_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+	preTn = read.table(pre_tn_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+	# Ability to switch between copy-ratio and log2 copy-ratio
+	if (log_input) {
+	    tnDat = 2^tn[,sample_name]
+	    preTnDat = 2^tn[,sample_name]
+	}else {
+	    tnDat = tn[,sample_name]
+	    preTnDat = tn[,sample_name]
+	}
+    tnContig = tn[,"contig"]
+    tnPos = tn[,"stop"]
+    preTnContig = preTn[,"contig"]
+    preTnPos = preTn[,"stop"]
+    preQc = QC(preTnDat)
+    postQc = QC(tnDat)
+
+    plot.fn = file.path( output_dir, paste(sample_name, "_FullGenome", ".png", sep=""))
+    png(plot.fn, 12, 7, units="in", type="cairo", res=300, bg="white")
+    chr.dat = SetUpPlot("Total copy ratio", 0, 4, "", TRUE)
+    TotalCopyRatioWithSegments(tnDat, tnPos, tnContig, seg, chr.dat)
+    dev.off()
+
+    pre_color="#3B5DFF"
+    post_color="#4FC601"
+    plot.fn = file.path( output_dir, paste(sample_name, "_Before_After_Lim4", ".png", sep=""))
+    png(plot.fn, 12, 7, units="in", type="cairo", res=300, bg="white")
+    par( mfrow=c(2,1), cex=0.75, las=1 )
+    SetUpPlot("Copy-Ratio", 0, 4, paste("Pre Tangent Normalization, pre-Qc = ", round(preQc, 3), sep=""), FALSE)
+    TotalCopyRatio(preTnDat, preTnPos, preTnContig, chr.dat, sample_name, pre_color)
+    SetUpPlot("Copy-Ratio", 0, 4, paste("Post Tangent Normalization, post-Qc = ", round(postQc, 3), ", dQc = ",
+        round(preQc-postQc,3), ", scaled_Qc = ", round((preQc-postQc)/preQc,3), sep=""), TRUE)
+    TotalCopyRatio(tnDat, tnPos, tnContig, chr.dat, sample_name, post_color)
+    dev.off()
+
+    plot.fn = file.path( output_dir, paste(sample_name, "_Before_After", ".png", sep=""))
+    png(plot.fn, 12, 7, units="in", type="cairo", res=300, bg="white")
+    par( mfrow=c(2,1), cex=0.75, las=1 )
+    SetUpPlot("Copy-Ratio", 0, max(tnDat), paste("Pre Tangent Normalization, pre-Qc = ", round(preQc, 3), sep=""), FALSE)
+    TotalCopyRatio(preTnDat, preTnPos, preTnContig, chr.dat, sample_name, pre_color)
+    SetUpPlot("Copy-Ratio", 0, max(preTnDat), paste("Post Tangent Normalization, post-Qc = ", round(postQc, 3),
+        ", dQc = ", round(preQc-postQc,3), ", scaled_Qc = ", round((preQc-postQc)/preQc,3), sep=""), TRUE)
+    TotalCopyRatio(tnDat, tnPos, tnContig, chr.dat, sample_name, post_color)
+    dev.off()
+}
+
+QC = function(dat) {
```

Comment:
I know this is the QC that we have always used, but really we should use median consecutive difference *divided by standard deviation*.  Otherwise, we would report a wonderful QC improvement if tangent normalization did nothing but rescale the data.  Scaling in this way isolates the noise reduction from the scale reduction.